File: CarePackage/Software/PlayStationAccessories.cs
namespace CarePackage.Software;

public class PlayStationAccessories : BaseSoftware
{
    public override string           Key            { get; set; } = "playstation-accessories";
    public override string           Name           { get; set; } = "PlayStation Accessories";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Peripheral;
    public override string           DownloadName   { get; set; } = "PlayStationAccessoriesInstaller.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = true;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.playstation_accessories;
    public override string           Homepage       { get; set; } = "https://controller.dl.playstation.net/controller/lang/en/2100004.html";
    
    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://fwupdater.dl.playstation.net/fwupdater/PlayStationAccessoriesInstaller.exe");
}

File: packages/app/src/services/software/definitions/playstationAccessories.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'playstation-accessories';
	public name = 'PlayStation Accessories';
	public category = [SoftwareCategory.Gaming, SoftwareCategory.Peripheral];
	public downloadName = 'PlayStationAccessoriesInstaller.exe';
	public requiresAdmin = true;
	public icon = 'playstation-accessories.png';
	public homepage = 'https://controller.dl.playstation.net/controller/lang/en/2100004.html';

	public async resolveDownloadUrl() {
		return ok('https://fwupdater.dl.playstation.net/fwupdater/PlayStationAccessoriesInstaller.exe');
	}
}
